Tests/InAppMessagingModuleSpec.swift

```diff
@@ -278,6 +278,29 @@ class InAppMessagingModuleSpec: QuickSpec {
                                         expect(campaignRepository.wasResetDataPersistenceCalled).to(beFalse())
                                     }
                             }
+
+                            it("will clear event list when user logs out or changes to another user") {
+                                [(aUser, nil), (aUser, IAMPreference()),
+                                 (aUser, IAMPreferenceBuilder().setUserId("userB").build())]
+                                    .forEach { prefA, prefB in
+                                        iamModule.registerPreference(prefA)
+                                        eventMatcher.wasClearNonPersistentEventsCalled = false // reset
+                                        iamModule.registerPreference(prefB)
+                                        expect(eventMatcher.wasClearNonPersistentEventsCalled).to(beTrue())
+                                    }
+                            }
+
+                            it("will not clear event list when user did not log out or change to another user") {
+                                [(nil, aUser), (IAMPreference(), aUser),
+                                 (nil, nil), (nil, IAMPreference()),
+                                 (IAMPreference(), nil), (IAMPreference(), IAMPreference())]
```

Comment:
Why is (IAMPreference(), nil) not considered log out? Cos it's an empty prefs object?

Reply:
Yeah. prefs can be `nil` empty with nil fields (IAMPreference()) and empty with empty fields (which I didn't test)... It all depends on how host app is using the SDK.